components/script/dom/navigator.rs

```diff
@@ -24,6 +27,13 @@ use crate::dom::window::Window;
 use crate::dom::xrsystem::XRSystem;
 use crate::script_runtime::JSContext;
 
+pub(super) fn hardware_concurrency() -> u64 {
```

Comment:
Why is this function public? Could it simply be moved into the body of `HardwareConcurrency(&self)`? Since the code is not repeated, I'm not sure an extra function buys us much unless I'm misunderstanding.

Reply:
It's also called in `components/script/dom/workernavigator.rs` - I split it out to another function to avoid duplicating the logic in the `Navigator` and `WorkerNavigator`.